index.js

```diff
@@ -1 +1,207 @@
+app.use(methodOverride('_method'));
+
+// Helper Function that converts request.body to an Iterable Obj
+const convertToIterableObj = (nonIterableObj) => {
+  const newObj = {};
+  // Convert request.body into an iterable object
+  Object.entries(nonIterableObj).forEach(([key, value]) => {
+    newObj[key] = value;
+  });
+  return newObj;
+};
+
+// Helper Function that checks the inputs for validation
+const performValidationOnRequestBody = (requestBodyObj, callback) => {
+  const inputValidationFeedbackObj = {};
+  let isFormValid = true;
+  const {
+    _, description, date_time, city, state, shape, duration, summary, __,
+  } = requestBodyObj;
+
+  if (description.length < 5) {
+    inputValidationFeedbackObj.description = 'Description is too short!';
+    isFormValid = false;
+  } else {
+    inputValidationFeedbackObj.description = 'ok';
+  }
+
+  if (shape.toLowerCase() !== 'sphere' && shape.toLowerCase() !== 'circle') {
+    inputValidationFeedbackObj.shape = 'Shape is not valid.';
+    isFormValid = false;
+  } else {
+    inputValidationFeedbackObj.shape = 'ok';
+  }
+  callback(inputValidationFeedbackObj, isFormValid);
+};
+
+// Route: Homepage - Shows all sightings, sortable by ID
+app.get('/', (request, response) => {
+  const orderOfSort = request.query.sortOrder;
+  read(FILENAME, (data, error) => {
+    if (error) {
+      response.sendStatus(500, error);
+      return;
+    }
+    data.sightings.sort((a, b) => {
+      if (orderOfSort === 'ascending') {
+        return a.id - b.id;
+      } if (orderOfSort === 'descending') {
+        return b.id - a.id;
+      }
+      return a.id - b.id;
+    });
+    response.render('allSightings', data);
+  });
+});
+
+// Route: Create form to input sighting
+app.get('/newSighting', (request, response) => {
+  read(FILENAME, (data, error) => {
+    if (error) {
+      response.sendStatus(500, error);
+      return;
+    }
+    response.render('submitNewSightingForm', data);
+  });
+});
+
+// Create a POST method / request to submit new sighting
+app.post('/newSighting', (request, response) => {
+  const requestBodyObj = convertToIterableObj(request.body);
+  // locally scoped boolean to track if current form is valid
+  let isThisFormValid = true;
+
+  // Perform input validation first
+  performValidationOnRequestBody(requestBodyObj, (feedbackMsg, areFieldsValid) => {
+    if (areFieldsValid === false) {
+      request.body.feedback = feedbackMsg;
+      isThisFormValid = false;
+      response.render('submitNewSightingForm', request.body);
+    }
+  });
+  if (isThisFormValid === false) {
+    return;
+  }
+
+  /**
+   * @param {FILENAME} contains all the data of the sightings to be read
+   * @param {key} the key to access in the file to be added
+   * @param {input} in this case, it is the iterated requestBodyObj
+   * @param {callback} performs a callback after taking in the data and error vars from add fn
+  */
```

Comment:
Thanks for trying JSDoc comments! Typically these JSDoc function-level comments are applied at function declarations/definitions, not function invocations. We wouldn't typically see JSDoc comments in the middle of a function definition, most commonly seen above a function declaration line, e.g. above where `add` is declared/defined.